### ISSUE TITLE
Crash in AFSDKEventCache

### ISSUE THREAD

**Issue by DagAgren** [AppsFlyerSDK/AppsFlyerFramework#193]
# Report

## SDK Version

6.4.2

## What happened?

After https://github.com/AppsFlyerSDK/AppsFlyerFramework/issues/187 was fixed we are still seeing this crash in AppsFlyerLib:

```
Crashed: com.appsflyer.serial
0  libobjc.A.dylib                0x4304 object_getMethodImplementation + 68
1  CoreFoundation                 0x2c82c _NSIsNSString + 60
2  Foundation                     0x8cdf8 _NSIsPList + 68
3  Foundation                     0x8cfc8 _NSIsPList + 532
4  Foundation                     0x8cfc8 _NSIsPList + 532
5  Foundation                     0xb2f34 -[NSDictionary(NSDictionary) writeToFile:atomically:] + 92
6  WoltClient Production          0x76e464 +[AFSDKEventCache cacheEvent:] + 4345291876
7  WoltClient Production          0x784c1c -[AppsFlyerLib executeRequestWithUrlString:eventOptions:parameters:completionHandler:] + 4345383964
8  WoltClient Production          0x782420 __71-[AppsFlyerLib callServerWithEventName:eventValues:options:completion:]_block_invoke_2.1031 + 4345373728
9  libdispatch.dylib              0x1c04 _dispatch_call_block_and_release + 32
10 libdispatch.dylib              0x3950 _dispatch_client_callout + 20
11 libdispatch.dylib              0xb0ac _dispatch_lane_serial_drain + 664
12 libdispatch.dylib              0xbc10 _dispatch_lane_invoke + 392
13 libdispatch.dylib              0x16318 _dispatch_workloop_worker_thread + 656
14 libsystem_pthread.dylib        0x11b0 _pthread_wqthread + 288
15 libsystem_pthread.dylib        0xf50 start_wqthread + 8
```

## Additional version information

This crash seem to have been around for a long time and is not recently introduced.

**Comment by ge-af** [comment]
Hi @DagAgren ,

Thank you for reaching out to us!

To provide optimal support, we would kindly ask you to submit a support ticket to support@appsflyer.com.
When submitting the ticket, please specify your AppsFlyer sign-up (account) email, app ID, reproduction steps, code snippets, logs, the links you used, and any additional relevant information.

**Comment by DagAgren** [comment]
Similarly to https://github.com/AppsFlyerSDK/AppsFlyerFramework/issues/192, it seems this crash was introduced in December 2020 when we changed from using the Cocoapod "Beta-AppsFlyerFramework" version 6.0.1.168, to "AppsFlyerFramework" version 6.1.0.

**Comment by ge-af** [comment]
Fix for this one was just released in v6.5.2 of the iOS SDK.

**Comment by VanQuvideo** [comment]
still crashing in v6.5.2
